plenum/server/message_handlers.py

```diff
@@ -131,8 +131,9 @@ def create(self, msg: Dict, **kwargs) -> Optional[PrePrepare]:
         return pp
 
     def requestor(self, params: Dict[str, Any]) -> Optional[PrePrepare]:
-        return self.node.replicas[params['inst_id']].getPrePrepare(
-            params['view_no'], params['pp_seq_no'])
+        if self.node.replicas[params['inst_id']].isPrimary:
+            return self.node.replicas[params['inst_id']].getPrePrepare(
+                params['view_no'], params['pp_seq_no'])
```

Comment:
Actually, we haven't to explicitly check if the replica is primary. It's sufficient just to search for the message in `replica.sentPrePrepares` collection only, not in both `replica.sentPrePrepares` and `replica.prePrepares` where `replica.getPrePrepare` searches.

Reply:
We can't do it because in this case catchup after view change will impossible. Primary node will have not PrePrepare in sentPrePrepares after view change.